rust/cadeau/src/xmf/vpx/encoder.rs

```diff
@@ -0,0 +1,219 @@
+use super::{VpxCodec, VpxError, VpxImage, VpxPacket};
+
+use xmf_sys::{
+    XmfVpxEncoder, XmfVpxEncoderConfig, XmfVpxEncoder_Create, XmfVpxEncoder_Destroy, XmfVpxEncoder_EncodeFrame,
+    XmfVpxEncoder_Flush, XmfVpxEncoder_FreeEncodedFrame, XmfVpxEncoder_GetEncodedFrame, XmfVpxEncoder_GetLastError,
+    XmfVpxEncoder_GetPacket,
+};
+
+pub struct VpxEncoderConfig(XmfVpxEncoderConfig);
+
+pub struct VpxEncoderConfigBuilder {
+    codec: Option<VpxCodec>,
+    width: Option<u32>,
+    height: Option<u32>,
+    bitrate: Option<u32>,
+    timebase_num: Option<i32>,
+    timebase_den: Option<i32>,
+    threads: Option<u32>,
+}
+
+pub struct VpxEncoder {
+    ptr: *mut XmfVpxEncoder,
+}
+
+impl VpxEncoder {
+    pub fn new(config: VpxEncoderConfig) -> Self {
+        // SAFETY: The way we build the config ensures that it is always valid.
+        let ptr = unsafe { XmfVpxEncoder_Create(config.0) };
+        Self { ptr }
+    }
```

Comment:
[This comment](https://github.com/Devolutions/cadeau/pull/55#discussion_r1845883264) can be applied to the encoder module too.